File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int n= (int)s.length(), lo=0, ans=0;
        vector<int> alp(26);
        for(int hi=0; hi<n;++hi){
            alp[s[hi]-'A']++;
            int clen=0;
            for(int i=0;i<26;++i){
                clen=max(clen, alp[i]);
            }
            while(hi-lo+1-clen>k){
                alp[s[lo++]-'A']--;
                clen=0;
                for(int i=0;i<26;++i){
                    clen=max(clen, alp[i]);
                }    
            }
            ans=max(ans, hi-lo+1);
        }
        return ans;
    }
};

File: 424-longest-repeating-character-replacement/longest-repeating-character-replacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        int ans=0;
        int[] alp = new int[26];
        int left=0;
        for(int right=0; right<s.length(); right++){
            char ch = s.charAt(right);
            alp[ch-'A']++;
            int maxm=0;
            for(int i=0;i<26;i++) maxm= Math.max(maxm,alp[i]);
            int len= right-left+1, diff= len-maxm;
            while(diff>k){
                alp[s.charAt(left++)-'A']--;
                for(int i=0;i<26;i++) maxm= Math.max(maxm,alp[i]);
                len= right-left+1; diff= len-maxm;
            }
            ans= Math.max(ans,len);
        }
        return ans;
    }
}